mautic/mautic
# Campaign events reschedules due to DST

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Campaign events can get stuck in an hourly rescheduling loop with the following circumstances:

The event was evaluated prior to a time change that falls back an hour to be executed at a time after the time change. E.g. date_triggered = 2021-10-24 17:00:00 and trigger_date = 2021-11-08 14:00:00.
The event is scheduled at a relative time period at a specific hour or between two hours.

#### Steps to test this PR:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a campaign with event which can be scheduled after 7th November 2021.
4. Add a contact in Eastern time zone where DST changes are applicable and add that contact to the segment which used in above campaign.
3. Change date_triggered date in mautic_campaign_lead_event_log to befoe DST 7th November 2021. and verify event get triggered at correct time
